### ISSUE TITLE
[Feature Request]: Asynchronous LZMA compression

### ISSUE THREAD

**Issue by coornio** [PCSX2/pcsx2#6186]
### Description

Currently, both in Wx and Qt, the LZMA compression is a slow method (as opposed to the near-instant Zstandard) that can take usually up to a couple seconds of pause for single frame, and easily dozens for longer durations of multi-frame dumps.

My suggestion, as per the title, is to consider an asynchronous compression thread. Let the dump be captured uncompressed, then a CPU thread not in use for emulation can be delegated to handle compression of the dump independently, without pausing the entire emulation in order to finish.

Granted, odd situations may arise, like someone performing big dumps back-to-back for some reason (meaning there's a need for a queue implementation), and the potential of running out of storage space (but that was already a "maybe?" concern even without suggestion, I have no idea), but all in all I believe this would be great for smoothing out the jarring pauses of LZMA without making it feel like the emulator froze up potentially scaring people.

### Reason

a

### Examples

a

**Comment by stenzek** [comment]
I don't really see the value of this to be honest. GS dumps are a debugging feature, zstd dumps can be taken in semi-realtime (10-30fps depending on the game). LZMA is just way too slow for larger dumps, we're talking minutes, and gigabytes of memory usage to buffer it all.

**Comment by coornio** [comment]
Fair enough, I just felt it might be useful to offset how slow it is in general. Reckon that LZMA might get the boot entirely at some point then?

**Comment by stenzek** [comment]
It can still be beneficial for creating dumps to upload to github, as the compression ratio is a fair bit higher (zstd can be too, but I chose the speedier mode for faster capturing).

**Comment by coornio** [comment]
That's my thoughts as well, but then you're faced with the dilemma: go with a MUCH faster compression to get a quick file to upload, or wait a long while by potentially repeating dumps due to tricky capture spot and keep freezing with each shot so that you can benefit from the smaller size? :P 

Sorta feels like doing it manually at this point might be faster still, at least you get a progress bar.

Now I said it jokingly *but a progress bar sounds nifty honestly*

**Comment by lightningterror** [comment]
Closing as trivial, and what sten said.
